### PR TITLE
Add tolowercase feature

### PR DESCRIPTION
**Pull request by akleineb**
Sometimes you want to allow urls like `t3n.de/camelCase`, which will automatically redirect to `t3n.de/camelcase`. This ensures that all backlinks/external links will return a valid page with a standardized path (lowerecased with trailing slash).

### CODE REVIEW

File: Classes/RoutingComponent.php [hunk 1]
```diff
@@ -41,26 +42,63 @@ class RoutingComponent extends \Neos\Flow\Mvc\Routing\RoutingComponent
     protected $configuration;
 
     /**
-     * Redirect automatically to the trailing slash url
+     * Redirect automatically to the trailing slash url or lowered url if activated
      */
     public function handle(ComponentContext $componentContext): void
     {
+        $trailingSlashIsEnabled = isset($this->configuration['enable']['trailingSlash']) ? $this->configuration['enable']['trailingSlash'] === true : true;
```
Comment (johannessteu) [inline Classes/RoutingComponent.php]: Else part should be `false`, right?

File: Tests/Unit/SlackServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    public function uriWithSpecialCharsDoesNotThrowAnException()
+    {
+        $routingComponent = new RoutingComponent();
+
+        $uri = new Uri('http://dev.local/äß&/');
+        $newUri = $routingComponent->handleToLowerCase($uri);
+        $newUri = $routingComponent->handleTrailingSlash($newUri);
+
+        $this->assertEquals($uri, $newUri);
+    }
```
Comment (johannessteu) [inline Tests/Unit/SlackServiceTest.php]: What about a test if the correct status code is set and if the `redirectIfNecessary` acts like we expect it to?

**Review by johannessteu — CHANGES_REQUESTED** [review]

File: Classes/RoutingComponent.php (hunk 1 above)
Comment (akleineb) [reply Classes/RoutingComponent.php]: true!

File: Tests/Unit/SlackServiceTest.php (hunk 2 above)
Comment (akleineb) [reply Tests/Unit/SlackServiceTest.php]: Good point, i have added another 2 tests.

**Comment by akleineb** [comment]
Cause this merge request is a little big bigger than expected before, i have adjsuted the readme.

**Review by johannessteu — APPROVED** [review]

**Pull request merged**